ADR/ADR036-add-solidqueue-to-forms-runner.md

```diff
@@ -0,0 +1,37 @@
+# ADR036: SolidQueue and ActiveJob for managing background processing
+
+Date: 2024-11-25
+
+## Status
+
+Approved
+
+## Context
+
+Forms will be using AWS SES to deliver emails to form processors with files attached to the emails. For processing of these emails, we looked at Sidekiq and Redis for managing a queue and to give us the ability to retry failed submissions. However with Redis, the details of a submission are ephemeral once a job is taken to be processed. Due to the way Sidekiq uses Redis in the free version is it possible for it to lose jobs if the process crashes. It is very important that we do not lose submissions, so this risk isn't acceptable to us.
```

Comment:
To understand the context better it would be helpful to have a sentence or list that explains what our requirements are. Then we can explaining the options we considered and why they do/don't work. For example:

> For our team to successfully process submission emails we must be able to:
> - retry sending the email on failed submissions
> - a queuing mechanism to send emails asynchronously  
> - no limit to message size in the queuing system
(I don't know if that's what we need... but I think that's why I'd find it clearer)

Reply:
Yeh I think this would be helpful. In particular we need both of the things you mentioned:

- A queueing mechanism to send emails asynchronously
- A way to retry the email on failed submissions, for example if an email bounces. 

In particular with bounced emails, we only hear about this some time after the email attempt. A notification is published to an SNS topic that we will subscribe an SQS queue to. Ideally we would automatically retry a number of times, but we will need to keep the data until we have successfully managed to send it on (within whatever limits our policies allow).

The message size limit requirement is slightly uncertain. If we think, for example, the SQS limit is fine for now - we'd like to know if it ever becomes not fine, and ideally not lose any user data in this case.

Reply:
Have incorporated some of this! Thank you